pennylane/qnodes/jacobian.py

```diff
@@ -23,25 +23,54 @@
 
 from .base import BaseQNode, QuantumFunctionError
 
+DEFAULT_STEP_SIZE = 0.3
```

Comment:
Out of curiosity: what was the heuristic that led to a value of 0.3?

Reply:
This was just based on some initial numerics - I think what's key is to get the right order of magnitude. We can then update the number if we have more confidence on what it is.

Would it be possible for the default to depend on the number of shots?

Reply:
Yep, we should definitely have the default depend on the number of shots once we know the relationship!